numpyro/distributions/continuous.py

```diff
@@ -1188,6 +1188,70 @@ def infer_shapes(
         return batch_shape, event_shape
 
 
+class CAR(Distribution):
+    arg_constraints = {
+        "loc": constraints.real_vector,
+        "alpha": constraints.unit_interval,
+        "tau": constraints.positive,
+        "W": constraints.positive_definite,
+    }
+    support = constraints.real_vector
+    reparametrized_params = [
+        "loc",
+        "alpha",
+        "tau",
+    ]
+
+    def __init__(
+        self,
+        loc=0.0,
+        alpha=None,
+        tau=None,
+        W=None,
+        validate_args=None,
+    ):
+        if jnp.ndim(loc) == 0:
+            (loc,) = promote_shapes(loc, shape=(1,))
+        # temporary append a new axis to loc
+        loc = loc[..., jnp.newaxis]
+        if W is not None:
+            loc, self.W = promote_shapes(loc, W)
+
+        batch_shape = lax.broadcast_shapes(jnp.shape(loc)[:-2], jnp.shape(self.W)[:-2])
+        event_shape = jnp.shape(self.W)[-1:]
+        self.loc = loc[..., 0]
+        self.alpha = alpha
+        self.tau = tau
+        super(CAR, self).__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
+        )
+
+    def sample(self, key, sample_shape=()):
+        # MVN with covariance matrix inv(tau * (D - alpha * W))
```

Comment:
I think you can use:
```
D = jnp.diagonal(self.W.sum(axis=-1), axis1=-2, axis2=-1)
tau = jnp.expand_dims(self.tau, (-2, -1))
alpha = jnp.expand_dims(self.alpha, (-2, -1))
P = tau * (D - alpha * self.W))
mvn = dist.MultivariateNormal(self.loc, precision_matrix=P)
return mvn.sample(key, sample_shape=sample_shape)
```

Reply:
`axis1=-2, axis2=-1` is very confusing. But I back your shape experience!

Reply:
We'll have to revisit if this works for sparse W

Reply:
We need to expand tau and alpha in case they have batch dimensions. E.g. you might want to have CAR priors for a batch of real vectors.

Reply:
`jnp.diagonal(W.sum(axis=-1), axis1=-2, axis2=-1)` with my test `W = jnp.array(
            [
                [0.0, 1.0, 1.0, 0.0],
                [1.0, 0.0, 0.0, 1.0],
                [1.0, 0.0, 0.0, 1.0],
                [0.0, 1.0, 1.0, 0.0],
            ]
        ) `

throws an error of `ValueError: axis 1 is out of bounds for array of dimension 1`. Any idea?

Reply:
Sorry, I wanted to convert a diagonal vector to a diagonal matrix there (the revert of the operator I used). I think we can use
```
D = self.W.sum(axis=-1, keepdims=True) * jnp.eye(self.W.shape[-1])
```